src/types/helperTypes.d.ts

```diff
@@ -161,3 +161,95 @@ export type DataSourceForPlaceable<P extends PlaceableObject> = P extends Placea
     ? D["_source"]
     : never
   : never;
+
+type EndValue = string | bigint | number | boolean | null | undefined;
```

Comment:
I would call this `PrimitiveValues` as that seems to be a more descriptive name. Though if it's meant to be any primitive value it should be `symbol` as well.